src/macros.js

```diff
@@ -95,6 +101,42 @@ defineMacro("\\TextOrMath", function(context) {
     }
 });
 
+// Basic support for global macro definitions:
+//     \gdef\macro{expansion}
+//     \gdef\macro#1{expansion}
+//     \gdef\macro#1#2{expansion}
+//     \gdef\macro#1#2#3#4#5#6#7#8#9{expansion}
+defineMacro("\\gdef", function(context) {
+    let arg = context.consumeArgs(1)[0];
+    if (arg.length !== 1) {
+        throw new ParseError("\\gdef's first argument must be a macro name");
+    }
+    const name = arg[0].text;
+    // Count argument specifiers, and check they are in the order #1 #2 ...
+    let numArgs = 0;
+    arg = context.consumeArgs(1)[0];
+    while (arg.length === 1 && arg[0].text === "#") {
+        arg = context.consumeArgs(1)[0];
+        if (arg.length !== 1) {
+            throw new ParseError(`Invalid argument number length "${arg.length}"`);
+        }
+        if (!(/^[1-9]$/.test(arg[0].text))) {
+            throw new ParseError(`Invalid argument number "${arg[0].text}"`);
+        }
+        numArgs++;
+        if (parseInt(arg[0].text) !== numArgs) {
+            throw new ParseError(`Argument number "${arg[0].text}" out of order`);
+        }
```

Comment:
Even though this check is sufficient, having both allows for more nuanced error report.  Nice!